From 24dd7e2098d78620fbb022fdd118a5df676a722c Mon Sep 17 00:00:00 2001
From: lunatickochiya <55731491+lunatickochiya@users.noreply.github.com>
Date: Sun, 28 Dec 2025 20:24:01 +0800
Subject: [PATCH] =?UTF-8?q?nftables-nojson=EF=BC=9Afix=20compile=20with=20?=
 =?UTF-8?q?fullcone?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

修复在编译nftables-nojson出现错误

src/expression.c:827:27: error: 'fullcone_expr_json' undeclared here (not in a function); did you mean 'fullcone_expr_clone'?
---
 .../nftables/patches/999-11-masq-fullcone-flag.patch      | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/package/network/utils/nftables/patches/999-11-masq-fullcone-flag.patch b/package/network/utils/nftables/patches/999-11-masq-fullcone-flag.patch
index 735e870198..cb4e70fdd5 100644
--- a/package/network/utils/nftables/patches/999-11-masq-fullcone-flag.patch
+++ b/package/network/utils/nftables/patches/999-11-masq-fullcone-flag.patch
@@ -75,7 +75,7 @@ diff --git a/src/expression.c b/src/expression.c
 index c0cb7f22..39d2821a 100644
 --- a/src/expression.c
 +++ b/src/expression.c
-@@ -803,6 +803,51 @@ struct expr *range_expr_alloc(const struct location *loc,
+@@ -803,6 +803,55 @@ struct expr *range_expr_alloc(const struct location *loc,
  	return expr;
  }
  
@@ -100,7 +100,11 @@ index c0cb7f22..39d2821a 100644
 +	.type		= EXPR_FULLCONE,
 +	.name		= "fullcone",
 +	.print		= fullcone_expr_print,
-+	.json		= fullcone_expr_json,
++#ifdef ENABLE_JSON
++    .json   = fullcone_expr_json,
++#else
++    .json   = NULL,
++#endif
 +	.clone		= fullcone_expr_clone,
 +	.destroy	= fullcone_expr_destroy,
 +};
-- 
2.43.0

